git/updateindex.go

```diff
@@ -129,9 +129,32 @@ func UpdateIndexFromReader(c *Client, opts UpdateIndexOptions, r io.Reader) (*In
 			return nil, fmt.Errorf("update-index --index-info variant 1 not implemented")
 		case 3:
 			switch len(spaces[1]) {
-			case 20:
```

Comment:
What happened to case 20? Should it produce an error about variant 3?

Reply:
It was always wrong.  A Sha1 is 20 bytes long,  which is 40 characters when written as a hex string. It just wasn't caught before because modes are never 20 characters either.